middleware/grpc/shared.go

```diff
@@ -39,6 +39,9 @@ func spanName(rti *stats.RPCTagInfo) string {
 
 func handleRPC(ctx context.Context, rs stats.RPCStats) {
 	span := zipkin.SpanFromContext(ctx)
+	if span.IsNoop() {
+		return
```

Comment:
in go I guess we don't need to apply the context (to ensure sampled false is carried down stream)?

Reply:
That is a good question. My idea was that even if sampled=false is propagated we still get a trace ID and client could choose to ignore that and continue the trace?

Reply:
we had problems in brave where we early exited instead of making the span part of the context on noop. I suppose here is not responsible for making the span.. something else did, hence you are checking isNoop... iotw I think you already addressed this concern.